test/e2e/common/node/pod_resize.go

```diff
@@ -894,8 +894,8 @@ func doPodResizeTests(f *framework.Framework) {
 
 			patchAndVerify := func(patchString string, expectedContainers []e2epod.ResizableContainerInfo, initialContainers []e2epod.ResizableContainerInfo, opStr string, isRollback bool) {
 				ginkgo.By(fmt.Sprintf("patching pod for %s", opStr))
-				patchedPod, pErr = f.ClientSet.CoreV1().Pods(newPod.Namespace).Patch(context.TODO(), newPod.Name,
-					types.StrategicMergePatchType, []byte(patchString), metav1.PatchOptions{})
+				patchedPod, pErr = f.ClientSet.CoreV1().Pods(newPod.Namespace).Patch(ctx, newPod.Name,
```

Comment:
Curious if there is a reason to test with `Patch` instead of `Resize` here.  I see some integration tests use `Resize`, and I see value to testing via all possible paths (`Resize`, `Patch` and `Update`)..

Reply:
Gotcha. A few other things to consider here:  Will developers be using these tests are references for best practices?  If so, what approach to we want to encourage?  Also, do we have sufficient test coverage for `Resize`?  Would converting some of these tests help improve coverage?

I mean these as genuine questions, I haven't spent the time to assess what the answers are, but wanted to bring these considerations up.

Reply:
> Will developers be using these tests are references for best practices? If so, what approach to we want to encourage?

I think `Patch` request is always better since it takes only required changes and avoids resource conflicts.

> Also, do we have sufficient test coverage for Resize? Would converting some of these tests help improve coverage?

e2e tests in this file are more to test kubelet behavior rather than k8s client behavior. integration tests might be better fit to test `Resize` and `Patch`. I have added [basic integration tests](https://github.com/kubernetes/kubernetes/pull/128266/files#diff-4bc3f052b24d529f60a80616320bed5cea00e447f65e5dee1511706d2f595d09R785) for both of them. The reason for this is because we have [unit tests](https://github.com/kubernetes/kubernetes/pull/128266/files#diff-7ee023b4401073292170227985c8ac98015db0c1980657c277a7cea011c24830R24821) covering all possible validation cases.

Reply:
> > Will developers be using these tests are references for best practices? If so, what approach to we want to encourage?
> 
> I think `Patch` request is always better since it takes only required changes and avoids resource conflicts.

Makes sense, I've [added a separate suggestion](https://github.com/kubernetes/kubernetes/pull/128266/files#r1815206956) based on this line of thinking.